### PR TITLE
Cmds pre

### PR DESCRIPTION
**Pull request by compiaffe**
# Aivero Pull Request

## Describe what this PR/MR is about, what does it add, fix or improve

Add a `cmdsPre` entry in the payload to separate commonly used conan commands in the conan usecase and allow running custom conan commands by overriding the default `cmds`

Part of https://github.com/aivero/internal-issues/issues/40
----

## PR Review Checklist

These are based on the a variation of the boy-scout-rules.

Please make sure all of these are covered and ticked:

### Unit level code smells. 

Does the code smell on a unit (i.e. function) level. Confirm that **all** the functions are:

- [ ] Short (max 15 lines)
- [ ] Simple
- [ ] Have a simple interface, no more than 4 arguments. (Argue why you couldn't simplify the interface)

### Bad comments

Check the inline documentation:

- [ ] Inline documentation that is not updated or in-sync with the code
- [ ] Function- or module-level documentation that doesn't reflect the current workings of the fn/mod.
- [ ] Overly extensive documentation - could the documented code be replaced with a function?

### General smell of code

Check that there are none of the below:

- [ ] Code in comments
- [ ] Dead code
- [ ] Long identifiers
- [ ] Magic constants
- [ ] No badly handled exceptions. Or in Rust, preferably no .expect or .unwrap, unless we are 100% sure the code will no fail.

### Docs

- [ ] Is there a Changelog.[adoc, md, txt]?
- [ ] Does the changelog contain an entry documenting the changes of this MR and is the proposed version [SemVer](https://semver.org/) complient?

### Tests

There needs to be tests, how much and which is up to the reviewer to point out.

- [ ] Is there at least a single new test?
- [ ] Is the new code covered by unit tests?

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -329,6 +336,7 @@ class Mode {
       return SelectMode.Command;
     }
     // TODO: add support for other modes
+    // TODO: Allow specifying a mode manually.
```
Comment (noverby) [inline src/main.ts:339]: We can just add a mode setting to devops.yml

**Review by noverby — APPROVED** [review]

File: src/main.ts (hunk 1 above)
Comment (compiaffe) [reply src/main.ts:339]: Yes, but are we handling this anywhere? Not in this function as far as I can see :)

Comment (noverby) [reply src/main.ts:339]: Nope

Comment (compiaffe) [reply src/main.ts:339]:
```suggestion
    // TODO:  Handle a manually specified mode.
```

File: src/main.ts [hunk 2]
```diff
@@ -90,7 +95,7 @@ class Mode {
     const folder = path.dirname(confPath);
     const name = path.basename(folder);
     const conf = YAML.parse(confRaw);
-    let ints: {}[] = [];
+    const ints: {}[] = [];
```
Comment (compiaffe) [inline src/main.ts:98]: @noverby There are plenty of these changes around: ESlint suggested them. Sounds a little counterintuitive to be able to .push  to a const, but if the linter says so...

Comment (compiaffe) [reply src/main.ts:98]: Linter also suggests using: 
```suggestion
    const ints: Record<string, unknown>[] = [];
```

Linter suggest this on all {} across the file...

Comment (noverby) [reply src/main.ts]: Please don't change the {}[] to Records. Ideally we should just use the interfaces.

Comment (compiaffe) [reply src/main.ts]: I've followed the suggestions

**Pull request merged**